### PR TITLE
Chore: Add Vite configuration and dependencies

### LINKED ISSUE #21: Project Bootstrap
**__Project Bootstrap__**
Ticket created in order to define and bootstrap the initial stage of the application.

**__Technical Considerations__**
As per discussed on our last call 10/05/2024, we decided to go with the following stack:

- **Module Bundler**: Vite
- **View Framework**: React
- **Language**: TypeScript
- **CSS Library**: [Tailwind CSS](https://tailwindcss.com) + [shadcn/ui](https://ui.shadcn.com)

**__Additional Considerations__**
TypeScript is going to be the main language, however it will be configured in a way that allow us to use JavaScripts too, so that developers that are not familiar with it can still colaborate.

A meeting will be scheduled soon to go over the setup. Anybody that wants to join the call feel free to contact me @feliperdamaceno

### PR DESCRIPTION
**Pull request by fmtabbara**
Closes #21

### CODE REVIEW

**Review by VriaA — APPROVED** [review]

File: .gitignore
Comment (feliperdamaceno) [inline .gitignore:1]: we should include .env files to be ignored:

```
# local env files
.env
.env.local
.env.development.local
.env.test.local
.env.production.local
```

File: .prettierrc
Comment (feliperdamaceno) [inline .prettierrc:1]: I would add the below to avoid issues, as some of us uses Mac/Windows/Linux:

```
"endOfLine": "lf"
```

File: package.json
Comment (feliperdamaceno) [inline package.json:1]: I like using this prettier plugin to sort imports, something to consider maybe?

https://github.com/trivago/prettier-plugin-sort-imports

File: public/vite.svg
Comment (feliperdamaceno) [inline public/vite.svg:1]: Unused assets could've been removed 🙂

File: tsconfig.json
Comment (feliperdamaceno) [inline tsconfig.json:1]: Forgot to include flag to allow using JS for members that don't know how to use it. Although I would avoid as much as possible.

```
"allowJs": true
```

**Review by feliperdamaceno — COMMENTED** [review]
I would include a file `.npmrc` with `save-exact=true`. In my work team, is standard to avoid using automatic package updated to avoid breaking changes. If we go on adding that, we've to remove all the `^` from the package versions.

Other than that is all good ✌🏻

**Pull request merged**